contracts/DelFiPrice.sol

```diff
@@ -62,7 +82,17 @@ contract DelFiPrice is OpenOracleView {
         }
 
         uint64[] memory sortedPrices = sort(postedPrices);
-        return sortedPrices[N / 2];
+
+        // if N is even, get the left and right medians and average them
+        if (N % 2 == 0) {
+            uint64 left = sortedPrices[(N / 2) - 1];
+            uint64 right = sortedPrices[N / 2];
+            require((left + right) >= left, "DelfiPrice::MedianPrice price addition overflow");
+            return (left + right) / 2;
```

Comment:
Alternatively we could use an intermediate uint or uint128 for left + right, since we know (left + right) / 2 will fit in uint64. Saves a tiny bit of gas but also might look cleaner 🤷‍♂